backend/routes/templates.go

```diff
@@ -35,10 +40,58 @@ func hashTemplateImage(pixelData []byte) string {
 	return hash.String()
 }
 
-func imageToPixelData(imageData []byte) []byte {
-	// TODO: Convert image data to pixel data using approximation
-	//       Output should be a byte array with color indexes
-	return []byte{0, 1, 1, 2, 2, 3}
+func imageToPixelData(imageData []byte) ([]byte, error) {
+	img, _, err := image.Decode(bytes.NewReader(imageData))
+	if err != nil {
+		return nil, err
+	}
+
+	palette := []color.Color{
```

Comment:
Can you load the color palette from `core.ArtPeaceBackend.CanvasConfig.Colors` and dynamically generate this array from that data?